root-common/usr/share/container-scripts/mysql/common.sh

```diff
@@ -91,17 +93,32 @@ function shutdown_local_mysql() {
   mysqladmin $admin_flags flush-privileges shutdown
 }
 
+# Extracts the initial temporary root password from the log file
+function get_autogenerated_root_password() {
+  AUTOGENERATED_ROOT_PASSWORD="$(cat $MYSQL_LOG_FILE | grep "temporary password" | awk '{print $NF}')"
+}
+
 # Initialize the MySQL database (create user accounts and the initial database)
 function initialize_database() {
   log_info 'Initializing database ...'
   if [[ "$MYSQL_VERSION" < "5.7" ]] ; then
     # Using --rpm since we need mysql_install_db behaves as in RPM
     log_and_run mysql_install_db --rpm --datadir=$MYSQL_DATADIR
   else
-    log_and_run ${MYSQL_PREFIX}/libexec/mysqld --initialize-insecure --datadir=$MYSQL_DATADIR --ignore-db-dir=lost+found
+    log_to_file_and_run ${MYSQL_PREFIX}/libexec/mysqld --initialize --datadir=$MYSQL_DATADIR --ignore-db-dir=lost+found
   fi
+
+  get_autogenerated_root_password
+  mysql_flags="$mysql_flags -p${AUTOGENERATED_ROOT_PASSWORD}"
+  admin_flags="--defaults-file=$MYSQL_DEFAULTS_FILE $mysql_flags"
+
   start_local_mysql "$@"
 
+  # This workaround is needed to avoid getting an 'expired password' error message
+  mysql --connect-expired-password $mysql_flags <<EOSQL
+    SET PASSWORD = PASSWORD('${AUTOGENERATED_ROOT_PASSWORD}');
```

Comment:
The expiration actually says that you should change the password asap, which we should do, rather than workaround this check.